src/compiler/checker.ts

```diff
@@ -12881,6 +12881,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         const typeParameter = getTypeParameterFromMappedType(type);
         const constraintType = getConstraintTypeFromMappedType(type);
         const nameType = getNameTypeFromMappedType(type.target as MappedType || type);
+        const isFilteringMappedType = nameType && isTypeAssignableTo(nameType, typeParameter);

```

Comment:
This seems like more work than the rest of the function does, so I'm worried about performance problems on mapped types with a nameType. I'll run the perf tests, but I don't know that any such mapped types are in there. @weswigham do you have an idea if this `isTypeAssignableTo` call is likely to be (relatively) slow here?

Reply:
Note that it's the same check that got used here:
https://github.com/microsoft/TypeScript/pull/48699/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R15745

That said - that other PR calls this just from within `getSimplifiedIndexedAccessType` and this is in the `resolveMappedTypeMembers` so the impact on the perf could still be different as it depends on how frequent are calls to those caller functions